test/connection.test.ts

```diff
@@ -1214,6 +1214,113 @@ describe('Connection', function () {
     })
   })
 
+  describe('Custom Fulfill Predicate', function () {
+    beforeEach(async function () {
+      this.clientPlugin.deregisterDataHandler()
+      this.serverPlugin.deregisterDataHandler()
+    })
+
+    it('use shouldFulfill callback to fulfill or reject packets', async function () {
+      const serverMoneySpy = sinon.spy()
+      let addressAndSecret: {
+        sharedSecret: Buffer
+        destinationAccount: string
+      }
+      let serverConn: Connection
+
+      /**
+       * Total to send: 100
+       * Max packet amount: 60
+       *
+       * Three packets with money should be received by the server
+       * and provided to `shouldFulfill`:
```

Comment:
I had to do some digging to figure out how why these are sequences 7, 8, 9
You could maybe mention that 1-6 are test packets.